packages/flutter/lib/src/widgets/overscroll_indicator.dart

```diff
@@ -221,15 +221,17 @@ class _GlowingOverscrollIndicatorState extends State<GlowingOverscrollIndicator>
     return NotificationListener<ScrollNotification>(
       onNotification: _handleScrollNotification,
       child: RepaintBoundary(
-        child: CustomPaint(
-          foregroundPainter: _GlowingOverscrollIndicatorPainter(
-            leadingController: widget.showLeading ? _leadingController : null,
-            trailingController: widget.showTrailing ? _trailingController : null,
-            axisDirection: widget.axisDirection,
-            repaint: _leadingAndTrailingListener,
-          ),
-          child: RepaintBoundary(
-            child: widget.child,
+        child: ClipRect(
```

Comment:
Adding the ClipRect here is probably not a good idea since clipping is expansive and in the vast majority of cases unnecessary. Unfortunately, your case where you have a BackdropFilter is one of the rare cases where you'd need clipping. This is because BackdropFilter behaves somewhat counterintuitively in where it applies the blur to (see https://github.com/flutter/flutter/pull/28017 where I try to add more docs to explain it better).

Instead of making everybody pay for the clip wether they need it or not (most won't) I'd suggest you add the clip just to your app. You should get the effect you want by applying a ClipRect around the BackdropFilter on this line in your test app: https://gist.github.com/magazmj/ada31013a9bffb805ea7d9c5f1b61820#file-main-dart-L35